packages/backend/src/peripherals/database/WithdrawableAssetRepository.test.ts

```diff
@@ -0,0 +1,227 @@
+      starkKey: StarkKey.fake(),
+      assetType: AssetHash.fake(),
+      tokenId: 1n,
+      assetId: AssetHash.fake(),
+      nonQuantizedAmount: 567000n,
+      quantizedAmount: 567n,
+      recipient: EthereumAddress.fake(),
+      ...override,
+    }
+  }
+
+  function fakeMintWithdrawalPerformedData(
+    override?: Partial<MintWithdrawData>
+  ): MintWithdrawData {
+    return {
+      type: 'MintWithdraw',
+      starkKey: StarkKey.fake(),
+      assetType: AssetHash.fake(),
+      nonQuantizedAmount: 678000n,
+      quantizedAmount: 678n,
+      assetId: AssetHash.fake(),
+      ...override,
+    }
+  }
+
+  describe(WithdrawableAssetRepository.prototype.add.name, () => {
+    it('adds records', async () => {
+      const fake1 = {
+        transactionHash: Hash256.fake(),
+        blockNumber: 123,
+        timestamp: Timestamp(123000),
+        data: fakeWithdrawalAllowedData(),
+      }
+      const id1 = await repository.add(fake1)
+
+      const fake2 = {
+        transactionHash: Hash256.fake(),
+        blockNumber: 123,
+        timestamp: Timestamp(123000),
+        data: fakeMintableWithdrawalAllowedData(),
+      }
+      const id2 = await repository.add(fake2)
+
+      const fake3 = {
+        transactionHash: Hash256.fake(),
+        blockNumber: 123,
+        timestamp: Timestamp(123000),
+        data: fakeAssetWithdrawalAllowedData(),
+      }
+      const id3 = await repository.add(fake3)
+
+      const fake4 = {
+        transactionHash: Hash256.fake(),
+        blockNumber: 123,
+        timestamp: Timestamp(123000),
+        data: fakeWithdrawalPerformedData(),
+      }
+      const id4 = await repository.add(fake4)
+
+      const fake5 = {
+        transactionHash: Hash256.fake(),
+        blockNumber: 123,
+        timestamp: Timestamp(123000),
+        data: fakeWithdrawalWithTokenIdPerformedData(),
+      }
+      const id5 = await repository.add(fake5)
+
+      const fake6 = {
+        transactionHash: Hash256.fake(),
+        blockNumber: 123,
+        timestamp: Timestamp(123000),
+        data: fakeMintWithdrawalPerformedData(),
+      }
+      const id6 = await repository.add(fake6)
+
+      const record1 = await repository.findById(id1)
+      const record2 = await repository.findById(id2)
+      const record3 = await repository.findById(id3)
+      const record4 = await repository.findById(id4)
+      const record5 = await repository.findById(id5)
+      const record6 = await repository.findById(id6)
+
+      expect(record1?.transactionHash).toEqual(fake1.transactionHash)
+      expect(record1?.blockNumber).toEqual(fake1.blockNumber)
+      expect(record1?.timestamp).toEqual(fake1.timestamp)
+      expect(record1?.data).toEqual(fake1.data)
+      expect(record2?.data).toEqual(fake2.data)
+      expect(record3?.data).toEqual(fake3.data)
+      expect(record4?.data).toEqual(fake4.data)
+      expect(record5?.data).toEqual(fake5.data)
+      expect(record6?.data).toEqual(fake6.data)
```

Comment:
Is there a reason why we are not doing this in a loop?

Reply:
I prefer to be more explicit in tests - adding a loop can introduce subtle bugs.